caffe2/onnx/torch_ops/defs.cc

```diff
@@ -5,17 +5,103 @@
 
 namespace ONNX_NAMESPACE {
 
-static const char* dummy_test_only_ver1_doc = R"DOC(
-A dummy op for verifying the build setup works, don't use me.
-)DOC";
+ONNX_PYTORCH_OPERATOR_SET_SCHEMA(
+    SparseLengthsSumFused8BitRowwise,
+    1,
+    OpSchema()
+        .SetDoc("Mirror Caffe2 SparseLengthsSumFused8BitRowwise operator")
+        .Input(0, "DATA", "data tensor", "T1")
+        .Input(1, "INDICES", "indices tensor", "T2")
+        .Input(2, "LENGTHS", "lengths tensor", "T2")
+        .Output(0, "output", "Output tensor", "T2")
+        .TypeConstraint(
+            "T1",
+            {"tensor(uin8)"},
+            "Constrain input data to uint8 tensors.")
+        .TypeConstraint(
+            "T2",
+            {"tensor(int8)",
+             "tensor(int16)",
+             "tensor(int32)",
+             "tensor(int64)",
+             "tensor(uint8)",
+             "tensor(uint16)",
+             "tensor(uint32)",
+             "tensor(uint64)"},
+            "Constrain index and length to integral tensors."));
+
+ONNX_PYTORCH_OPERATOR_SET_SCHEMA(
+    SparseLengthsSum,
+    1,
+    OpSchema()
+        .SetDoc("Mirror Caffe2 SparseLengthsSum operator")
+        .Input(0, "DATA", "data tensor", "T1")
+        .Input(1, "INDICES", "indices tensor", "T2")
+        .Input(2, "LENGTHS", "lengths tensor", "T2")
+        .Output(0, "output", "Output tensor", "T1")
+        .TypeConstraint(
+            "T1",
+            {"tensor(float16)", "tensor(float)", "tensor(double)"},
+            "Constrain input and output types to float tensors.")
+        .TypeConstraint(
+            "T2",
+            {"tensor(int8)",
+             "tensor(int16)",
+             "tensor(int32)",
+             "tensor(int64)",
+             "tensor(uint8)",
+             "tensor(uint16)",
+             "tensor(uint32)",
+             "tensor(uint64)"},
+            "Constrain index and length to integral tensors."));
+
+ONNX_PYTORCH_OPERATOR_SET_SCHEMA(
+    DotProduct,
+    1,
+    OpSchema()
+        .SetDoc("Mirror Caffe2 DotProduct operator")
+        .Input(0, "X", "Input 1 tensor", "T")
+        .Input(1, "Y", "Input 2 tensor", "T")
+        .Output(0, "Z", "Output tensor", "T")
+        .TypeConstraint(
+            "T",
+            {"tensor(float16)", "tensor(float)", "tensor(double)"},
+            "Constrain input and output types to float tensors."));
+
+ONNX_PYTORCH_OPERATOR_SET_SCHEMA(
+    FCTransposed,
+    1,
+    OpSchema()
+        .SetDoc("Mirror Caffe2 FCTransposed operator")
+        .Input(0, "X", "Input tensor", "T")
+        .Input(1, "W", "Weight tensor", "T")
+        .Input(2, "B", "Bias tensor", "T")
+        .Output(0, "Z", "Output tensor", "T")
+        .TypeConstraint(
+            "T",
+            {"tensor(float16)", "tensor(float)", "tensor(double)"},
+            "Constrain input and output types to float tensors."));
+
+ONNX_PYTORCH_OPERATOR_SET_SCHEMA(
+    BatchMatMul,
```

Comment:
This should be equivalent to ONNX's MatMul

Reply:
ONNX matmul supports batch?

Reply:
Yep it behaves like [numpy's matmul](https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.matmul.html).

Reply:
Could we keep it as BatchMatMul, as Glow is implementing BatchMatMul and right now, for these ops I would like a 1-1 mapping between Glow and C2 to help with the debug. We can further reduce it into other primitives once we have a baseline.

Reply:
The onnx -> caffe2 backend actually translates ONNX's MatMul to c2's BatchMatMul already, so you probably don't need this?